app/src/main/java/org/schabi/newpipe/settings/preferencesearch/PreferenceSearchAdapter.java

```diff
@@ -0,0 +1,91 @@
+package org.schabi.newpipe.settings.preferencesearch;
+
+import android.text.TextUtils;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+import androidx.recyclerview.widget.RecyclerView;
+
+import org.schabi.newpipe.R;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Consumer;
+
+class PreferenceSearchAdapter
+        extends RecyclerView.Adapter<PreferenceSearchAdapter.PreferenceViewHolder> {
+    private List<PreferenceSearchItem> dataset = new ArrayList<>();
+    private Consumer<PreferenceSearchItem> onItemClickListener;
+
+    @NonNull
+    @Override
+    public PreferenceSearchAdapter.PreferenceViewHolder onCreateViewHolder(
+            @NonNull final ViewGroup parent,
+            final int viewType
+    ) {
+        return new PreferenceViewHolder(
+            LayoutInflater
+                .from(parent.getContext())
+                .inflate(R.layout.settings_preferencesearch_list_item_result, parent, false));
+    }
+
+    @Override
+    public void onBindViewHolder(
+            @NonNull final PreferenceSearchAdapter.PreferenceViewHolder holder,
+            final int position
+    ) {
+        final PreferenceSearchItem item = dataset.get(position);
+
+        holder.title.setText(item.getTitle());
+
+        if (TextUtils.isEmpty(item.getSummary())) {
+            holder.summary.setVisibility(View.GONE);
+        } else {
+            holder.summary.setVisibility(View.VISIBLE);
+            holder.summary.setText(item.getSummary());
+        }
+
+        if (TextUtils.isEmpty(item.getBreadcrumbs())) {
+            holder.breadcrumbs.setVisibility(View.GONE);
+        } else {
+            holder.breadcrumbs.setVisibility(View.VISIBLE);
+            holder.breadcrumbs.setText(item.getBreadcrumbs());
+        }
+
+        holder.itemView.setOnClickListener(v -> {
+            if (onItemClickListener != null) {
+                onItemClickListener.accept(item);
+            }
+        });
+    }
+
+    void setContent(final List<PreferenceSearchItem> items) {
+        dataset = new ArrayList<>(items);
+        this.notifyDataSetChanged();
+    }
+
+    @Override
+    public int getItemCount() {
+        return dataset.size();
+    }
+
+    void setOnItemClickListener(final Consumer<PreferenceSearchItem> onItemClickListener) {
+        this.onItemClickListener = onItemClickListener;
+    }
+
+    static class PreferenceViewHolder extends RecyclerView.ViewHolder {
+        final TextView title;
+        final TextView summary;
+        final TextView breadcrumbs;
```

Comment:
I tried to implement it with ``SettingsPreferencesearchListItemResultBinding`` but there are 2 problems:
1. https://stackoverflow.com/questions/68586270/viewbinding-with-java11-androidstudio-always-shows-error-but-runs-without-any
2. ``itemView`` is not present because the binding does not extend ``RecyclerView.ViewHolder``

Any idea how to fix ``2.``?

Reply:
For 2. you can just use `binding.getRoot()`: if you check in the source code of `RecyclerView.ViewHolder` the value that gets assigned to `itemView' is just the view passed to the constructor of the `ViewHolder`, i.e. the root view

Reply:
This diff works:
```diff
diff --git a/app/src/main/java/org/schabi/newpipe/settings/preferencesearch/PreferenceSearchAdapter.java b/app/src/main/java/org/schabi/newpipe/settings/preferencesearch/PreferenceSearchAdapter.java
index 527a4a595..e3114f81e 100644
--- a/app/src/main/java/org/schabi/newpipe/settings/preferencesearch/PreferenceSearchAdapter.java
+++ b/app/src/main/java/org/schabi/newpipe/settings/preferencesearch/PreferenceSearchAdapter.java
@@ -4,12 +4,11 @@ import android.text.TextUtils;
 import android.view.LayoutInflater;
 import android.view.View;
 import android.view.ViewGroup;
-import android.widget.TextView;
 
 import androidx.annotation.NonNull;
 import androidx.recyclerview.widget.RecyclerView;
 
-import org.schabi.newpipe.R;
+import org.schabi.newpipe.databinding.SettingsPreferencesearchListItemResultBinding;
 
 import java.util.ArrayList;
 import java.util.List;
@@ -22,37 +21,35 @@ class PreferenceSearchAdapter
 
     @NonNull
     @Override
-    public PreferenceSearchAdapter.PreferenceViewHolder onCreateViewHolder(
+    public PreferenceViewHolder onCreateViewHolder(
             @NonNull final ViewGroup parent,
             final int viewType
     ) {
-        return new PreferenceViewHolder(
-            LayoutInflater
-                .from(parent.getContext())
-                .inflate(R.layout.settings_preferencesearch_list_item_result, parent, false));
+        return new PreferenceViewHolder(SettingsPreferencesearchListItemResultBinding.inflate(
+                LayoutInflater.from(parent.getContext()), parent, false));
     }
 
     @Override
     public void onBindViewHolder(
-            @NonNull final PreferenceSearchAdapter.PreferenceViewHolder holder,
+            @NonNull final PreferenceViewHolder holder,
             final int position
     ) {
         final PreferenceSearchItem item = dataset.get(position);
 
-        holder.title.setText(item.getTitle());
+        holder.binding.title.setText(item.getTitle());
 
         if (TextUtils.isEmpty(item.getSummary())) {
-            holder.summary.setVisibility(View.GONE);
+            holder.binding.summary.setVisibility(View.GONE);
         } else {
-            holder.summary.setVisibility(View.VISIBLE);
-            holder.summary.setText(item.getSummary());
+            holder.binding.summary.setVisibility(View.VISIBLE);
+            holder.binding.summary.setText(item.getSummary());
         }
 
         if (TextUtils.isEmpty(item.getBreadcrumbs())) {
-            holder.breadcrumbs.setVisibility(View.GONE);
+            holder.binding.breadcrumbs.setVisibility(View.GONE);
         } else {
-            holder.breadcrumbs.setVisibility(View.VISIBLE);
-            holder.breadcrumbs.setText(item.getBreadcrumbs());
+            holder.binding.breadcrumbs.setVisibility(View.VISIBLE);
+            holder.binding.breadcrumbs.setText(item.getBreadcrumbs());
         }
 
         holder.itemView.setOnClickListener(v -> {
@@ -64,6 +61,7 @@ class PreferenceSearchAdapter
 
     void setContent(final List<PreferenceSearchItem> items) {
         dataset = new ArrayList<>(items);
+        // noinspection NotifyDataSetChanged
         this.notifyDataSetChanged();
     }
 
@@ -77,15 +75,11 @@ class PreferenceSearchAdapter
     }
 
     static class PreferenceViewHolder extends RecyclerView.ViewHolder {
-        final TextView title;
-        final TextView summary;
-        final TextView breadcrumbs;
-
-        PreferenceViewHolder(final View v) {
-            super(v);
-            title = v.findViewById(R.id.title);
-            summary = v.findViewById(R.id.summary);
-            breadcrumbs = v.findViewById(R.id.breadcrumbs);
+        final SettingsPreferencesearchListItemResultBinding binding;
+
+        PreferenceViewHolder(final SettingsPreferencesearchListItemResultBinding binding) {
+            super(binding.getRoot());
+            this.binding = binding;
         }
     }
 }
```